src/Datadog.Trace/HttpOverStreams/HttpMessage.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Text;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.HttpOverStreams
+{
+    internal abstract class HttpMessage
+    {
+        private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<HttpMessage>();
+        private static readonly UTF8Encoding Utf8Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+
+        public HttpMessage(HttpHeaders headers, IHttpContent content)
+        {
+            Headers = headers;
+            Content = content;
+        }
+
+        public HttpHeaders Headers { get; }
+
+        public IHttpContent Content { get; }
+
+        public int? ContentLength => int.TryParse(Headers.GetValue("Content-Length"), out int length) ? length : (int?)null;
+
+        public string ContentType => Headers.GetValue("Content-Type");
+
+        public Encoding GetContentEncoding()
+        {
+            if (ContentType == null)
+            {
+                return null;
+            }
+
+            if (string.Equals("application/json", ContentType, StringComparison.InvariantCultureIgnoreCase))
+            {
+                // Default
+                return Utf8Encoding;
+            }
+
+            // text/plain; charset=utf-8
+            string[] pairs = ContentType?.Split(';');
+
+            foreach (string pair in pairs)
+            {
+                string[] parts = pair.Split('=');
+
+                if (parts.Length == 2 && string.Equals(parts[0].Trim(), "charset", System.StringComparison.OrdinalIgnoreCase))
+                {
+                    switch (parts[1].Trim())
```

Comment:
I don't know how the encoding is picked by the sender, but iso-8859-1 is commonly used in Europe

Reply:
I verified that this works within an app service deployed in Central France.
Is this a reasonable test?
Should I also deploy to a VM zoned in Europe somewhere?